Library/Homebrew/rubocops/extend/formula_cop.rb

```diff
@@ -316,8 +316,12 @@ def component_precedes?(first_node, next_node)
       end
 
       # Check if negation is present in the given node
-      def negated?(node)
-        method_called?(node, :!)
+      def expression_negated?(node)
+        if !node.parent.nil? && node.parent.method_name.equal?(:!)
```

Comment:
```
return false if node.parent.nil?
return false unless node.parent.method_name.equal?(:!)
```
and then you can skip the `return true` and indented `if`. In general checking for "bad" input and exiting early is a nice practice.

Reply:
@MikeMcQuaid Updated the PR.